### PR TITLE
browser detection component

### PR DESCRIPTION
**Pull request by shayanbinary**

### CODE REVIEW

File: app/browser-detection.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        cqA.href = 'http://outdatedbrowser.com/';
+        cqA.innerHTML = 'Update Browser';
+        cqNotSupported.appendChild(cqA);
+        
+        if (cqRoot.nextSibling) {
+            cqRoot.parentNode.insertBefore(cqNotSupported, cqRoot.nextSibling);
+        } else {
+            cqRoot.parentNode.appendChild(cqNotSupported);
+        }
+        cqRoot.parentNode.removeChild(cqRoot);
```
Comment (shayanbinary) [inline app/browser-detection.js]: sometimes we show `not supported` page because of CSS or any critical issue, so as I didn't find any safe way to warranty stop running rest of codes, I decide to stop it by removing `<div id="root" />`.

File: app/browser-detection.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    }
+
+    window.onerror = function(message, source, lineno, colno, error){
+        if (
+                (
+                    message.indexOf('[mobx] MobX 5+ requires Proxy objects') !== -1 ||
+                    message.indexOf('not supported browser') !== -1
```
Comment (shayanbinary) [inline app/browser-detection.js]: check error messages, perhaps a simple error happens that is not related to our browser support check

File: app/index.jsx [hunk 3]
```diff
@@ -124,21 +125,25 @@ class App extends Component {
         const { settings, isConnectionOpened, symbol } = this.state;
 
         return (
-            <SmartChart
-                id={chartId}
-                symbol={symbol}
+            <SupportDetection
```
Comment (shayanbinary) [inline app/index.jsx]: as decouple detection code perhaps I should remove this. agree?

File: app/browser-detection.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        var os = browser.os;
+
+        if (
+            SupportedBrowsers[os] && 
+            SupportedBrowsers[os][name] !== undefined) {
+            if(version < SupportedBrowsers[os][name]){
+                throw new Error('not supported browser');
+            }
+        } else {
+            alert('Please update your browser to the latest version.');
```
Comment (shayanbinary) [inline app/browser-detection.js]: because I don't have access to `notification` component so the only way is to `alert` the message

Comment (brucebinary) [reply app/browser-detection.js]: @shayanbinary why are you using `alert`? We have a design from the UX team to display some page is the browser is not supported. Use that instead

File: app/index.jsx (hunk 3 above)
Comment (brucebinary) [reply app/index.jsx]: yep

File: app/browser-detection.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
+function browserDetect(rootEleId, SupportedBrowsers){
```
Comment (brucebinary) [inline app/browser-detection.js]: what's this script here. Did you wrote this?

File: webpack.config.js [hunk 6]
```diff
@@ -93,6 +93,10 @@ const config = {
         }),
         new MiniCssExtractPlugin({ filename: 'smartcharts.css' }),
         new StyleLintPlugin(),
+        new CopyWebpackPlugin([
```
Comment (shayanbinary) [inline webpack.config.js]: @brucebinary Can I leave this piece of code here or I should put in under `if (production) {` ?

**Comment by ashkanx** [comment]
@shayanbinary conflicts

**Comment by shayanbinary** [comment]
@ashkanx Done.
